modules/oceanbase/src/main/java/org/testcontainers/oceanbase/OceanBaseCEContainer.java

```diff
@@ -0,0 +1,131 @@
+
+    private static final String DEFAULT_DATABASE_NAME = "test";
+
+    private String tenantName = DEFAULT_TEST_TENANT_NAME;
+
+    private String driverClassName = "com.mysql.cj.jdbc.Driver";
+
+    public OceanBaseCEContainer(String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    public OceanBaseCEContainer(DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+
+        addExposedPorts(SQL_PORT, RPC_PORT);
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return driverClassName;
+    }
+
+    @Override
+    public String getJdbcUrl() {
+        return getJdbcUrl(DEFAULT_DATABASE_NAME);
+    }
+
+    public String getJdbcUrl(String databaseName) {
+        String additionalUrlParams = constructUrlParameters("?", "&");
+        String prefix = driverClassName.contains("mysql") ? "jdbc:mysql://" : "jdbc:oceanbase://";
+        return prefix + getHost() + ":" + getMappedPort(SQL_PORT) + "/" + databaseName + additionalUrlParams;
+    }
+
+    @Override
+    public String getDatabaseName() {
+        return DEFAULT_DATABASE_NAME;
+    }
+
+    @Override
+    public String getUsername() {
+        return DEFAULT_USERNAME + "@" + tenantName;
+    }
+
+    @Override
+    public String getPassword() {
+        return DEFAULT_PASSWORD;
+    }
+
+    @Override
+    protected String getTestQueryString() {
+        return "SELECT 1";
+    }
+
+    /**
+     * Set the non-system tenant to be created for testing.
+     *
+     * @param tenantName the name of tenant to be created
+     * @return this
+     */
+    public OceanBaseCEContainer withTenant(String tenantName) {
+        if (StringUtils.isEmpty(tenantName)) {
+            throw new IllegalArgumentException("Tenant name cannot be null or empty");
+        }
+        if (SYSTEM_TENANT_NAME.equals(tenantName)) {
+            throw new IllegalArgumentException("Tenant name cannot be " + SYSTEM_TENANT_NAME);
+        }
+        this.tenantName = tenantName;
+        return self();
+    }
+
+    /**
+     * Set the driver class name.
+     *
+     * @param driverClassName the driver class name
+     * @return this
+     */
+    public OceanBaseCEContainer withDriverClassName(String driverClassName) {
+        if (StringUtils.isEmpty(driverClassName)) {
+            throw new IllegalArgumentException("Driver class name cannot be null or empty");
+        }
+        if (!driverClassName.contains("mysql") && !driverClassName.contains("oceanbase")) {
+            throw new IllegalArgumentException("Driver class name should contains 'mysql' or 'oceanbase'");
+        }
+        this.driverClassName = driverClassName;
+        return self();
+    }
+
+    @Override
+    protected void configure() {
+        if (!DEFAULT_TEST_TENANT_NAME.equals(tenantName)) {
+            withEnv("OB_TENANT_NAME", tenantName);
+        }
+    }
```

Comment:
according to the [docs](https://github.com/oceanbase/oceanbase/wiki/Deploy_with_Docker_CN#%E7%8E%AF%E5%A2%83%E5%8F%98%E9%87%8F), the default value for `OB_TENAN_NAME` is `test`. So we can just have

```suggestion
    @Override
    protected void configure() {
        withEnv("OB_TENANT_NAME", tenantName);
    }
```

Reply:
User can also use 'withEnv' method to set it manually, in this case if we do not check the value, there may be a mistake.

Reply:
I don't fully understand because the default is `test` and the `tenantName = DEFAULT_TEST_TENANT_NAME;` where `DEFAULT_TEST_TENANT_NAME = "test"`. Unless, the right one should use `SYSTEM_TENANT_NAME` instead

```java
if (!SYSTEM_TENANT_NAME.equals(tenantName)) {
    withEnv("OB_TENANT_NAME", tenantName);
}
```

Once we resolve this comment I can proceed and merge it

Reply:
I'm considering this situation:  someone use the container class but not set the tenant name with given method `withTenant`,  like `new OceanBaseCEContainer("xxx").withEnv("OB_TENANT_NAME", "tc");`. In this case, as the tenantName field is still 'test', a 'test' tenant will be created, not user defined value 'tc'.

I'm not sure if we should deal with it. WDYT?

Reply:
I think that given there is a default value in the image itself.  The `withTenant` method and the `withEnv` in `configure` method can be omitted. If for some reason there is a need to modify it then `new OceanBaseCEContainer(...).withEnv("OB_TENANT_NAME", "something");` that would a similar experience like running `docker run` command. Otherwise, it would be a little bit confusing.

Reply:
The tenantName field here is required to construct the username used in jdbc connection, so I did not remove the `withTenant` method in previous cleanup. I reconsidered about it and maybe use the default value with necessary description is better.

Reply:
Updated.

Reply:
I just noticed we were missing a test for this use case and also think a little bit more about it. In the `configure` method we can add `getEnvMap().computeIfAbsent("OB_TENAN_NAME", k -> tenantName != null ? tenantName : DEFAULT_TENANT_NAME)` or just `getEnvMap().computeIfAbsent("OB_TENAN_NAME", k -> tenantName)` and we can keep `withTenantName`. LMK if this can help to the use case described.

Reply:
Yes you are right, we can resolve it by using the envMap, but I think it's not necessary for now.

At present no other `withXXX` method is provided in this container class and all env options are set by `withEnv`, I think it's good and clean. Lets keep it like this.